ssh-credentials dependency of support plugin is not actually optional

### Your checklist for this issue

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [ ] Jenkins version

- [ ] Plugin version

- [ ] OS

<!--
Put an `x` into the [ ] to show you have filled the information below
Describe your issue below
-->

### Description

https://github.com/jenkinsci/configuration-as-code-plugin/blob/727c976d137461f146b301f302d1552ca81de75e/support/pom.xml#L45 declares an optional dependency.

https://github.com/jenkinsci/configuration-as-code-plugin/blob/727c976d137461f146b301f302d1552ca81de75e/support/src/main/java/io/jenkins/plugins/casc/support/credentials/DirectEntryPrivateKeySourceConfigurator.java#L15 is a nonoptional extension point.

Should be made `@Extension(optional = true)`

This is likely the cause of:

````
Apr 20, 2019 1:35:54 AM hudson.ExtensionFinder$GuiceFinder$SezpozModule configure
WARNING: Failed to load io.jenkins.plugins.casc.support.credentials.DirectEntryPrivateKeySourceConfigurator
java.lang.LinkageError: Failed to resolve class io.jenkins.plugins.casc.support.credentials.DirectEntryPrivateKeySourceConfigurator
        at hudson.ExtensionFinder$GuiceFinder$SezpozModule.resolve(ExtensionFinder.java:507)
        at hudson.ExtensionFinder$GuiceFinder$SezpozModule.configure(ExtensionFinder.java:527)
        at com.google.inject.AbstractModule.configure(AbstractModule.java:62)
        at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
        at com.google.inject.spi.Elements.getElements(Elements.java:110)
        at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104)
        at com.google.inject.Guice.createInjector(Guice.java:96)
        at com.google.inject.Guice.createInjector(Guice.java:73)
        at hudson.ExtensionFinder$GuiceFinder.<init>(ExtensionFinder.java:285)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.lang.Class.newInstance(Class.java:442)
        at net.java.sezpoz.IndexItem.instance(IndexItem.java:181)
        at hudson.ExtensionFinder$Sezpoz._find(ExtensionFinder.java:715)
        at hudson.ExtensionFinder$Sezpoz.find(ExtensionFinder.java:690)
        at hudson.ClassicPluginStrategy.findComponents(ClassicPluginStrategy.java:335)
        at hudson.ExtensionList.load(ExtensionList.java:380)
        at hudson.ExtensionList.ensureLoaded(ExtensionList.java:318)
        at hudson.ExtensionList.getComponents(ExtensionList.java:183)
        at jenkins.model.Jenkins$6.onInitMilestoneAttained(Jenkins.java:1127)
        at jenkins.InitReactorRunner$1.onAttained(InitReactorRunner.java:83)
        at org.jvnet.hudson.reactor.ReactorListener$Aggregator.lambda$onAttained$3(ReactorListener.java:102)
        at org.jvnet.hudson.reactor.ReactorListener$Aggregator.run(ReactorListener.java:109)
        at org.jvnet.hudson.reactor.ReactorListener$Aggregator.onAttained(ReactorListener.java:102)
        at org.jvnet.hudson.reactor.Reactor$1.run(Reactor.java:177)
        at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.TypeNotPresentException: Type com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey$DirectEntryPrivateKeySource not present
        at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:117)
        at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
        at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at sun.reflect.generics.visitor.Reifier.reifyTypeArguments(Reifier.java:68)
        at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:138)
        at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
        at sun.reflect.generics.repository.ClassRepository.getSuperclass(ClassRepository.java:90)
        at java.lang.Class.getGenericSuperclass(Class.java:777)
        at hudson.ExtensionFinder$GuiceFinder$SezpozModule.resolve(ExtensionFinder.java:488)
        ... 31 more
Caused by: java.lang.ClassNotFoundException: com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey$DirectEntryPrivateKeySource
        at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1374)
        at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1327)
        at jenkins.util.AntClassLoader.loadClass(AntClassLoader.java:1080)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
        ... 39 more
````

--- a/support/src/main/java/io/jenkins/plugins/casc/support/credentials/DirectEntryPrivateKeySourceConfigurator.java
+++ b/support/src/main/java/io/jenkins/plugins/casc/support/credentials/DirectEntryPrivateKeySourceConfigurator.java
@@ -12,7 +12,7 @@ import io.jenkins.plugins.casc.model.Scalar;
 /**
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
-@Extension
+@Extension(optional = true)
 public class DirectEntryPrivateKeySourceConfigurator extends DataBoundConfigurator<DirectEntryPrivateKeySource> {
 
     public DirectEntryPrivateKeySourceConfigurator() {

